File: src/main/java/org/myfintech/payment/repository/BaseRepository.java
package org.myfintech.payment.repository;

import org.myfintech.payment.entity.AbstractEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.NoRepositoryBean;

@NoRepositoryBean
public interface BaseRepository<T extends AbstractEntity<ID>, ID> extends JpaRepository<T, ID> {
    // Define shared custom methods if needed
}

File: src/main/kotlin/org/myfintech/payment/repository/BaseRepository.kt
package org.myfintech.payment.repository

import org.myfintech.payment.entity.AbstractEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.NoRepositoryBean

@NoRepositoryBean
interface BaseRepository<T : AbstractEntity<ID>, ID : Any> : JpaRepository<T, ID> {
    // You can define shared custom repository methods here if needed
}